File: strategy/php/strategies/PayPalPayment.php
<?php

class PayPalPayment implements PaymentStrategy {
    private $email;

    public function __construct($email) {
        $this->email = $email;
    }

    public function pay($amount) {
        echo "Pagando $" . $amount . " con PayPal\n";
        echo "Email de PayPal: " . $this->email . "\n";
        // Lógica de procesamiento de PayPal
    }
}

File: strategy/python/strategies/paypal_payment.py
from interfaces.payment_strategy import PaymentStrategy

class PayPalPayment(PaymentStrategy):
    def __init__(self, email):
        self.email = email

    def pay(self, amount):
        print(f"Pagando ${amount} con PayPal")
        print(f"Email de PayPal: {self.email}")
        # Lógica de procesamiento de PayPal